publish/api.py

```diff
@@ -34,14 +34,26 @@ def publish(self, request, *args, **kwargs):
         data = serializer.validated_data
         return Response(publish_dandiset.delay(**data).id)
 
-    @action(detail=True, methods=['GET'], url_path='(?P<version>[^/.]+)')
+    @action(detail=True, methods=['GET'], url_path='(?P<version>[.0-9]+)')
     def version(self, request, dandi_id, version):
         queryset = self.filter_queryset(self.get_queryset())
         filter_kwargs = {'dandi_id': dandi_id, 'version': version}
         obj = get_object_or_404(queryset, **filter_kwargs)
         serializer = DandisetSerializer(obj)
         return Response(serializer.data)
 
+    @action(detail=True, methods=['GET'], url_path='versions')
+    def versions(self, request, dandi_id):
+        queryset = self.filter_queryset(self.get_queryset())
+        filter_kwargs = {'dandi_id': dandi_id}
+        get_object_or_404(queryset, **filter_kwargs)
```

Comment:
If there are multiple `Dandiset` with the same `dandi_id`, then `get_object_or_404` will [raise `MultipleObjectsReturned`](https://docs.djangoproject.com/en/3.0/topics/http/shortcuts/).

I think its logically coherent to say that if `not Dandiset.objects.filter({'dandi_id': dandi_id}).exists()`, then a 404 should be raised. Otherwise, you can list the versions of the matching Dandisets.

Note, to be even more efficient, apparently you should use `len()` instead of `.count()`: https://docs.djangoproject.com/en/3.0/ref/models/querysets/#count on a reused queryset. I'd add a brief code comment explaining this fact though.